artichoke-backend/src/extn/core/enumerable/enumerable.rb

```diff
@@ -0,0 +1,450 @@
+      elsif block
+        max = val if block.call(val, max).positive?
+        min = val if block.call(val, min).negative?
+      else
+        max = val if (val <=> max).positive?
+        min = val if (val <=> min).negative?
+      end
+    end
+    [min, max]
+  end
+
+  def minmax_by(&block)
+    return to_enum :minmax_by unless block
+
+    max = nil
+    max_cmp = nil
+    min = nil
+    min_cmp = nil
+    first = true
+
+    self.each do |*val|
+      if first
+        max = min = val.__svalue
+        max_cmp = min_cmp = block.call(*val)
+        first = false
+      elsif (cmp = block.call(*val)) > max_cmp
+          max = val.__svalue
+          max_cmp = cmp
+      elsif (cmp = block.call(*val)) < min_cmp
+          min = val.__svalue
+          min_cmp = cmp
+      end
+    end
+    [min, max]
+  end
+
+  def none?(pat=NONE, &block)
+    if pat != NONE
+      self.each do |*val|
+        return false if pat === val.__svalue # rubocop:disable Style/CaseEquality # rubocop:disable Style/CaseEquality
+      end
+    elsif block
+      self.each do |*val|
+        return false if block.call(*val)
+      end
+    else
+      self.each do |*val|
+        return false if val.__svalue
+      end
+    end
+    true
+  end
+
+  def one?(pat=NONE, &block)
+    count = 0
+    if pat!=NONE
+      self.each do |*val|
+        count += 1 if pat === val.__svalue # rubocop:disable Style/CaseEquality
+        return false if count > 1
+      end
+    elsif block
+      self.each do |*val|
+        count += 1 if block.call(*val)
+        return false if count > 1
+      end
+    else
+      self.each do |*val|
+        count += 1 if val.__svalue
+        return false if count > 1
+      end
+    end
+
+    count == 1 ? true : false
+  end
+
+  def all?(pat=NONE, &block)
+    if pat != NONE
+      self.each do |*val|
+        return false unless pat === val.__svalue # rubocop:disable Style/CaseEquality
+      end
+    elsif block
+      self.each{|*val| return false unless block.call(*val)}
+    else
+      self.each{|*val| return false unless val.__svalue}
+    end
+    true
+  end
+
+  def any?(pat=NONE, &block)
+    if pat != NONE
+      self.each do |*val|
+        return true if pat === val.__svalue # rubocop:disable Style/CaseEquality
+      end
+    elsif block
+      self.each{|*val| return true if block.call(*val)}
+    else
+      self.each{|*val| return true if val.__svalue}
+    end
+    false
+  end
+
+  def each_with_object(obj, &block)
+    return to_enum(:each_with_object, obj) unless block
+
+    self.each {|*val| block.call(val.__svalue, obj) }
+    obj
+  end
+
+  def reverse_each(&block)
+    return to_enum :reverse_each unless block
+
+    ary = self.to_a
+    i = ary.size - 1
+    while i>=0
+      block.call(ary[i])
+      i -= 1
+    end
+    self
+  end
+
+  def cycle(nval = nil, &block)
+    return to_enum(:cycle, nval) unless block
+
+    n = nil
+
+    if nval.nil?
+      n = -1
+    else
+      n = nval.__to_int
```

Comment:
Methods that begin with an underscore (`_`) are mruby internal methods and are not part of Ruby Core. Can we reimplement this logic with methods in Ruby Core? Maybe `to_int` or `to_i`?